pkg/controller/registry/reconciler/reconciler.go

```diff
@@ -160,5 +168,25 @@ func Pod(source *v1alpha1.CatalogSource, name string, image string, saName strin
 			ServiceAccountName: saName,
 		},
 	}
+
+	// Set priorityclass if its annotation exists
+	if prio, ok := annotations[CatalogPriorityClassKey]; ok && prio != "" {
+		pod.Spec.PriorityClassName = prio
+	}
+
+	// Add PodSpec hash
+	// This hash info will be used to detect PodSpec changes
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	labels[PodHashLabelKey] = hashPodSpec(pod.Spec)
```

Comment:
I'm trying to wrap my head around whether this check should live further up in this functions control flow as the `labels` variable is provided as a parameter to the function. Is there any reason we'd want to perform these checks _before_ we build up the Pod type, as it looks like we blindly assign the parameter to the corresponding field in the metadata struct? Just thinking aloud and trying to determine if there's some edge case I'm missing here, besides we're potentially making duplicate calls.

Reply:
I'm not sure what you mean by commenting on this line specifically given there is no check here, just a label addition. This `Pod()` is where we create the Pod template for every registry pod so it is definite the place to add labels/annotations if needed.

Reply:
Sorry, I meant we might want to perform this `if labels == nil` check at the top of the function, as the `labels` variable is provided as a parameter to the Pod function, not that we would move that check outside of this function. We blindly assign the metadata.Labels to that variable when building up the Pod type definition earlier in the function.

Reply:
The default value of a map is a nil, so even if `labels` is nil, assigning to a nil to the pod metadata is the same as omitting the labels altogether. So I don't see why we need to check it earlier. I have to check it later because I will assign something to the map and you cannot assign to nil map which will lead to nil panic. Hence, the check is there.

Reply:
Yeah, I'm not super concerned about that anymore as I don't think there are any edge cases to worry about. I think this is more my preference at this point that we perform any upfront manipulations of the function parameters before we assign them to the Pod resource for readability.